### PR TITLE
Build KubeVela legacy chart

### LINKED ISSUE #528: [Feature] automatically generate chart for legacy k8s version
Now we have sent our chart to oss bucket, we can support legacy k8s version chart.

refer to https://github.com/crossplane/oam-kubernetes-runtime/issues/286

### PR DESCRIPTION
**Pull request by zzxwill**
Also pushed it to oss bucket
To fix #528

### CODE REVIEW

File: legacy/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+Currently, the main blocker is KubeVela uses CRD v1, while those old Kubernetes versions don't support CRD v1.
+So we generate v1beta1 CRD here for convenience. But we have no guarantee that KubeVela core will support the
+legacy Kubernetes versions. 
+
+Follow the instructions in [README](../README.md) to create a namespace like `vela-system` and add the OAM Kubernetes
+Runtime helm repo.
+
+```
+$ kubectl create namespace vela-system
+$ helm repo add kubevela https://kubevelacharts.oss-cn-hangzhou.aliyuncs.com/core
```
Comment (wonderflow) [inline legacy/README.md:15]: please fix the url

File: legacy/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+$ helm install -n vela-system vela-core-legacy kubevela/vela-core-legacy
+```
+
+If you'd like to install an older version of the legacy chart, use `helm search` to choose a proper chart version.
+```
+$ helm search repo vela-core-legacy --devel -l
+  NAME                                           	CHART VERSION   	APP VERSION     	DESCRIPTION
+  kubevela/vela-core-legacy                     	0.0.1             	0.1                 A Helm chart for OAM Kubernetes Resources Controller...
+  kubevela/vela-core-legacy                     	0.0.3             	0.3                 A Helm chart for OAM Kubernetes Resources Controller...
+  kubevela/vela-core-legacy                     	0.3.1-5.g11e1894	0.3.1-5.g11e1894	A Helm chart for OAM Kubernetes Resources Controller...
```
Comment (wonderflow) [inline legacy/README.md:30]: how do we generate the suffix?

File: legacy/README.md (hunk 1 above)
Comment (zzxwill) [reply legacy/README.md:15]: repo 加入了 artifacthub.io，依然要是用原来的 repo url https://artifacthub.io/packages/helm/kubevela/vela-core?modal=install

File: .github/workflows/registry.yml [hunk 3]
```diff
@@ -87,23 +91,30 @@ jobs:
       - name: Tag helm chart image
         run:  |
           version=${{ steps.get_version.outputs.VERSION }}
-          sed -i "s/latest/$version/g" charts/vela-core/values.yaml
+          sed -i "s/latest/$version/g" $HELM_CHART/values.yaml
```
Comment (wonderflow) [inline .github/workflows/registry.yml:94]: do we also need to sed the legacy chart for this version?

File: .github/workflows/registry.yml [hunk 4]
```diff
@@ -87,23 +91,30 @@ jobs:
...
           number=${version#"v"}
-          sed -i "s/0.1.0/$number/g" charts/vela-core/Chart.yaml
+          sed -i "s/0.1.0/$number/g" $HELM_CHART/Chart.yaml
+          sed -i "s/0.1.0/$number/g" $LEGACY_HELM_CHART/Chart.yaml
+      - name: Prepare legacy chart
+        run: |
+          rsync -r $LEGACY_HELM_CHART $HELM_CHARTS_DIR
+          rsync -r $HELM_CHART/* $LEGACY_HELM_CHART --exclude=Chart.yaml --exclude=crds
```
Comment (wonderflow) [inline .github/workflows/registry.yml:101]: move it before `tag helm chart image`, that can be easily maintained

File: legacy/README.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+```
+
+If you'd like to install an older version of the legacy chart, use `helm search` to choose a proper chart version.
+```
+$ helm search repo vela-core-legacy --devel -l
+  NAME                     	CHART VERSION	APP VERSION	DESCRIPTION
+  kubevela/vela-core-legacy	0.2          	0.2        	A Helm chart for legacy KubeVela core Controlle...
+  kubevela/vela-core-legacy	0.0.1        	0.1        	A Helm chart for legacy KubeVela core Controlle...
+
+$ helm install -n vela-system kubevela-legacy kubevela/vela-core-legacy --version 0.0.1 --devel
```
Comment (wonderflow) [inline legacy/README.md:31]: do we really need the flag `--devel` in our current chart regisrty?

File: .github/workflows/registry.yml (hunk 3 above)
Comment (zzxwill) [reply .github/workflows/registry.yml:94]: You are right. As changed `values.yaml` will be synchronized to the legacy chart, so I didn't change it directly.
```
      - name: Prepare legacy chart
        run: |
          rsync -r $LEGACY_HELM_CHART $HELM_CHARTS_DIR
          rsync -r $HELM_CHART/* $LEGACY_HELM_CHART --exclude=Chart.yaml --exclude=crds
```

File: .github/workflows/registry.yml (hunk 4 above)
Comment (zzxwill) [reply .github/workflows/registry.yml:101]: I moved it to the end of `tag helm chart image`. But your suggestion is also right.

File: legacy/README.md (hunk 5 above)
Comment (zzxwill) [reply legacy/README.md:31]: No, we can omit the flag.

**Review by wonderflow — APPROVED** [review]
LGTM

**Pull request merged**